File: DSA/Arrays/Code/6_ArrayDeletion.java
import java.util.Arrays;
import java.util.ArrayList;

// Array Deletion in Java
// Remove an element from a specific position

public class ArrayDeletion {
    
    static void deleteElement(int[] arr, int n, int position) {
        // Shift elements to the left
        for (int i = position; i < n - 1; i++) {
            arr[i] = arr[i + 1];
        }
    }
    
    static void displayArray(int[] arr, int n) {
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
    
    public static void main(String[] args) {
        // Method 1: Using array
        int[] arr = {10, 20, 30, 40, 50};
        int n = 5;
        
        System.out.print("Original array: ");
        displayArray(arr, n);
        
        // Delete from beginning
        deleteElement(arr, n, 0);
        n--;
        System.out.print("After deleting first element: ");
        displayArray(arr, n);
        
        // Method 2: Using ArrayList
        ArrayList<Integer> list = new ArrayList<Integer>();
        list.add(10);
        list.add(20);
        list.add(30);
        list.add(40);
        list.add(50);
        
        System.out.println("Original ArrayList: " + list);
        
        // Delete at position
        list.remove(0);
        System.out.println("After removing element at position 0: " + list);
        
        // Delete by value
        list.remove(Integer.valueOf(30));
        System.out.println("After removing value 30: " + list);
        
        // Delete from end
        list.remove(list.size() - 1);
        System.out.println("After removing last element: " + list);
    }
}

File: DSA/Arrays/Code/6_ArrayDeletion.py
# Array Deletion in Python
# Remove an element from a specific position in a list

arr = [10, 20, 30, 40, 50]

print("Original array:", arr)

# Method 1: Delete using del statement
arr1 = [10, 20, 30, 40, 50]
del arr1[0]
print("After deleting first element (del):", arr1)

# Method 2: Delete using pop()
arr2 = [10, 20, 30, 40, 50]
removed = arr2.pop(0)
print("After popping first element (pop):", arr2)
print("Removed element:", removed)

# Method 3: Delete from middle
arr3 = [10, 20, 30, 40, 50]
arr3.pop(2)
print("After deleting element at position 2:", arr3)

# Method 4: Delete from end
arr4 = [10, 20, 30, 40, 50]
arr4.pop()
print("After deleting last element:", arr4)

# Method 5: Delete using remove() - removes by value
arr5 = [10, 20, 30, 40, 50]
arr5.remove(30)
print("After removing value 30:", arr5)

# Method 6: Delete multiple elements
arr6 = [10, 20, 30, 40, 50]
del arr6[1:3]  # Delete elements from index 1 to 2
print("After deleting elements from index 1 to 2:", arr6)

# Method 7: Filter out elements
arr7 = [10, 20, 30, 40, 50]
arr7 = [x for x in arr7 if x != 30]
print("After filtering out 30:", arr7)
